### ISSUE TITLE
Guest Authors

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress-Authors#627]
This is the author page for a Guest Author: /author/mathias-clasen

![Screen Shot 2022-04-04 at 4 08 16 PM](https://user-images.githubusercontent.com/3868028/161623757-42da8862-1488-440e-9a28-dabeca60b278.png)

If they are in first place in the author order, everything is OK:
<img width="348" alt="Screen Shot 2022-04-04 at 4 10 20 PM" src="https://user-images.githubusercontent.com/3868028/161623834-311748f6-8286-4ac5-862f-b93c33940970.png">

However, a bug is introduced if a mapped user is in first place:
<img width="362" alt="Screen Shot 2022-04-04 at 4 10 59 PM" src="https://user-images.githubusercontent.com/3868028/161623886-fe66440d-9678-493a-9312-9ebf87575d14.png">

If the Guest Author is not in first place, the Mapped User shows in the Author page:
![Screen Shot 2022-04-04 at 4 11 20 PM](https://user-images.githubusercontent.com/3868028/161623971-7aae9022-9304-40b1-9ffe-9343b98e6671.png)

**Comment by stevejburge** [comment]
@andergmartins I think we saw a bug like this before?

**Comment by ojopaul** [comment]
@stevejburge i'm wondering if this could be related to the function that output the title? Can you test and update me if this happen in default theme so i can check the function to make sure it's correct.

Right now in my testing, i'm not able to recreate the issue as Guest author name is shown irrespective of the author list order.

**Comment by andergmartins** [comment]
@stevejburge @ojopaul this is a known issue when a theme is not prepared to show our Authors, but only the native WP author.

For guest authors, it is a special case. Every post has a `post_author` attribute that accepts a user ID. Our plugin set authors as taxonomy terms. But we will fill that attribute with the first user ID for the authors of the post. Guest authors do not have a user ID, so we use the user ID of the next author, recursively until we find one that is mapped to an user. If all them are guest authors, we use the user ID of the current user (who created/edited the post when setting the authors).

For displaying author information, we use a custom class we implement, `Author`, and we use that on the themes we have adapted. All the themes that do not use it, they will try to bring information from the user found on the post's `post_author` attribute.

We have a lot of filters on the plugin that try to fix that when someone search for the author information... but depends on the way the theme is coded, if it manually grab the user id, and  print its information, we can not know the exact context, so we can't change that information to show the correct user, without creating a child theme and adapting the required templates.

**Comment by andergmartins** [comment]
Btw, this can happens on some page builders too.

**Comment by stevejburge** [comment]
@andergmartins So should change this into documentation instead of a bug report?

"If this happens, you need a child theme ..."

**Comment by andergmartins** [comment]
@stevejburge yes, I think that is the correct label for that

**Comment by andergmartins** [comment]
@stevejburge yes, that is a good approach. It is not a bug on the plugin, but a limitation on the interaction with 3rd party themes

**Comment by stevejburge** [comment]
This is the page we'll use for more details: https://publishpress.com/knowledge-base/troubleshooting-author-pages/
